src/Groups/action.jl

```diff
@@ -267,3 +267,43 @@ function on_indeterminates(f::Nemo.MPolyElem, s::PermGroupElem)
 end
 
 ^(f::Nemo.MPolyElem, p::PermGroupElem) = on_indeterminates(f, p)
+
+
+#############################################################################
+##
+##  natural stabilizers in permutation groups
+
+function stabilizer(G::PermGroup, pnt::T) where T <: Oscar.IntegerUnion
+    return Oscar._as_subgroup(G, GAP.Globals.Stabilizer(G.X, GAP.GapObj(pnt), GAP.Globals.OnPoints))
+end
+
+function stabilizer(G::PermGroup, pnt::Vector{T}) where T <: Oscar.IntegerUnion
+    return Oscar._as_subgroup(G, GAP.Globals.Stabilizer(G.X, GAP.GapObj(pnt), GAP.Globals.OnTuples))
+end
+
+function stabilizer(G::PermGroup, pnt::Set{T}) where T <: Oscar.IntegerUnion
+    pnt = [x for x in pnt]
+    sort!(pnt)
+    return Oscar._as_subgroup(G, GAP.Globals.Stabilizer(G.X, GAP.GapObj(pnt), GAP.Globals.OnSets))
+end
+
+
+#############################################################################
+##
+##  natural stabilizers in matrix groups
+
+function stabilizer(G::MatrixGroup{ET,MT}, pnt::AbstractAlgebra.Generic.FreeModuleElem{ET}) where {ET,MT}
+    img = GAP.GapObj(G.ring_iso.(pnt.v))[1]
+    return Oscar._as_subgroup(G, GAP.Globals.Stabilizer(G.X, img, GAP.Globals.OnRight))
+end
+
+function stabilizer(G::MatrixGroup{ET,MT}, pnt::Vector{AbstractAlgebra.Generic.FreeModuleElem{ET}}) where {ET,MT}
+    img = GAP.GapObj([GAP.GapObj(G.ring_iso.(x.v))[1] for x in pnt])
```

Comment:
Shouldn't this be OnTuples? Or maybe it works because we are essentially doing matrix multiplication here? Alas, I wonder if that might prevent certain optimizations for OnTuples to kick in (namely one can compute a tuple stabilizer by iterating regular stabilizer computations over the tuple entries)

Reply:
Yes, the code was correct because of the interpretation as matrix multiplication.
And yes, there is in fact one situation in the GAP library where an `OnTuples` stabilizer is computed iteratively.
(The code is meanwhile looking differently, this question does not arise anymore.)